cpp/src/writer/tsfile_writer.cc

```diff
@@ -163,10 +172,12 @@ int TsFileWriter::register_timeseries(
     return register_timeseries(device_id, ms, false);
 }
 
-int TsFileWriter::register_timeseries(const std::string &device_id,
+int TsFileWriter::register_timeseries(const std::string &device_path,
                                       MeasurementSchema *measurement_schema,
                                       bool is_aligned) {
-    DeviceSchemaIter device_iter = schemas_.find(device_id);
+    std::shared_ptr<IDeviceID> device_id =
+        std::make_shared<PlainDeviceID>(device_path);
+    DeviceSchemasMapIter device_iter = schemas_.find(device_id);
```

Comment:
PlainDeviceID is not for the tree model, it is DEPRECATED,

Reply:
fixed